src/crimson/os/seastore/object_data_handler.cc

```diff
@@ -617,19 +618,30 @@ struct overwrite_plan_t {
    * original extent into at most three parts: origin-left, part-to-be-modified
    * and origin-right.
    */
-  void evaluate_operations() {
+  void evaluate_operations(Transaction& t) {
     auto actual_write_size = get_pins_size();
     auto aligned_data_size = get_aligned_data_size();
     auto left_ext_size = get_left_extent_size();
     auto right_ext_size = get_right_extent_size();
 
+    auto can_merge = [](Transaction& t, paddr_t paddr) {
+      CachedExtentRef ext;
+      if (paddr.is_relative() || paddr.is_delayed()) {
+	  return true;
+      } else if (t.get_extent(paddr, &ext) ==
```

Comment:
This looks expensive and unnecessary to lookup the cache here. Probably it can be avoided by associating the state of the refering extent to the pin.

I think it is fine for now, but worth an additional comment: 
`// FIXME: there is no need to lookup the cache if the pin can be associated with the extent state`